src/Pager.js

```diff
@@ -435,18 +426,37 @@ export default class Pager<T: Route> extends React.Component<Props<T>> {
   ]);
 
   _translateX = block([
-    call([this._index], ([value]) => {
-      this._currentIndexValue = value;
-    }),
+    onChange(
+      this._index,
+      call([this._index], ([value]) => {
+        this._currentIndexValue = value;
+
+        // Without this check, the pager can go to an infinite update <-> animate loop for sync updates
+        if (value !== this.props.navigationState.index) {
+          // If the index changed, and previous animation has finished, update state
+          this.props.onIndexChange(value);
+
+          this._pendingIndexValue = value;
+
+          // Force componentDidUpdate to fire, whether user does a setState or not
+          // This allows us to detect when the user drops the update and revert back
+          // It's necessary to make sure that the state stays in sync
+          this.forceUpdate();
+        }
+      })
+    ),
     // Conditionally listen for changes in the position value
     // The position value can changes a lot in short time
     // So we only add a listener when necessary to avoid extra overhead
     cond(
       this._isListening,
-      onChange(
-        this._position,
-        call([this._position], this._handlePositionChange)
-      )
+      onChange(this._position, [
+        cond(
+          eq(this._throttler, 0),
+          call([this._position], this._handlePositionChange)
+        ),
+        set(this._throttler, modulo(add(this._throttler, 1), THROTTLE_FACTOR)),
```

Comment:
Maybe we want to set `this._throttler` to zero the beginning to make it sure that listener will be invoked at least once at the gesture.

Reply:
Do you mean I should set it to 0 when the gesture starts?